invoices/mixins.py

```diff
@@ -1,166 +1,152 @@
-from django.http import HttpResponse, HttpResponseRedirect
-from django.shortcuts import render, redirect, get_object_or_404
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404
 from django.conf import settings
-from django.contrib.auth.mixins import PermissionRequiredMixin
-
-from reportlab.pdfbase import pdfmetrics 
-from reportlab.pdfbase.ttfonts import TTFont
-from reportlab.lib.enums import TA_JUSTIFY
-from reportlab.lib.pagesizes import letter
-from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, Image
-from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
-from reportlab.lib.units import inch
-from reportlab.pdfgen import canvas
-from reportlab.lib.utils import ImageReader
 
-from invoices.models import Invoice
 
+from invoices.models import Invoice
+from reportlab.platypus import Image
+from reportlab.pdfgen import canvas
 from io import BytesIO
 from io import StringIO
-
 from PIL import Image
-
-from datetime import datetime, date 
-import os, errno,io
+from datetime import date 
+import os, errno, io
 
 
 
 class PdfMixin(object):
+    """pdf functionality
+    """
+    def invoice_directory(self,_id, filename):
+        """Directory to save pdf
+        """
+        directory = '{media}{invoices}/{id}/'.format(media=settings.MEDIA_ROOT,invoices='invoices',id=_id)
```

Comment:
if python  < 3.5, you can do this. `"{}invoices/{}/".format(settings.MEDIA_ROOT, _id) 
if python 3.6, you can do this. `f"{settings.MEDIA_ROOT}/invoices/{_id}"`